### ISSUE TITLE
Migrate from py2exe

### ISSUE THREAD

**Issue by seanbudd** [nvaccess/nvda#18449]
### Related issues, PRs or discussions

https://github.com/py2exe/py2exe/issues/215
https://github.com/py2exe/py2exe/issues/191

### What is the current state of the codebase?

We use py2exe to bundle NVDA for the installer.

### Why are changes required?

We are stuck on Python 3.11 due to py2exe not supporting 3.12+.
The work required to fix py2exe is complicated, and the project has not had commits to it in 8 months.
It appears to be abandoned, and it seems likely that 3.12+ support won't happen.

### What technical changes are required?

Migrating to [pyInstaller](https://pyinstaller.org/en/stable/) - which does a very similar job

Alternatively, consider migrating to Nuitka. Nuitka compiles python to native code when creating the exe. This might have to be a longer term project though, as it would likely remove our current python add-on API.

### Are the proposed technical changes API breaking?

Yes

### Are there potential risks or issues with the proposed implementation?

pyInstaller is a well maintained project, and seems to have become the norm, over py2exe.

Nuitka or a similar python compiler might be something to explore once our new sandboxed add-on API is established, but moving to it would be quite drastic and completely break our current add-on framework.

**Comment by cary-rowen** [comment]
For me, if NVDA does not have an add-on system, or the permissions of add-ons are reduced to the extreme, it may be time to say goodbye to NVDA, and I hope there won’t be such a day.

**Comment by seanbudd** [comment]
@cary-rowen - there is no proposal here to do that

**Comment by seanbudd** [comment]
Experiments done in #18718 made compiling with nuitka infeasible. It is way too slow for a project as large as us. I think we can hope to gain better performance from the future of GIL-less python, UIA remote operations, and Rustifying where useful
